src/OpenTelemetry.Collector.Dependencies/AzureSdkDiagnosticListener.cs

```diff
@@ -60,6 +63,13 @@ public override void OnStartActivity(Activity current, object valueValue)
                 .SetCreateChild(false)
                 .SetSampler(this.sampler);
 
+            var links = LinksPropertyFetcher.Fetch(valueValue) as IEnumerable<Activity> ?? Array.Empty<Activity>();
+
+            foreach (var link in links)
+            {
+                spanBuilder.AddLink(Link.FromActivity(link));
```

Comment:
I believe Link .FromActivity does not work correctly - it assumes Activity is started (id format is set) and also uses Activity SpanId. In our case (which seems typical for links) Activity ParentSpanId is one that we care about and Activity is not started.

I'll try to fix it today.

Reply:
What does starting an activity change in terms of Ids?

Reply:
it generates new span id, it also determines format id during start as far as i remember

Reply:
we set id format in the producer to always be W3C but I see your point with span id.

Reply:
I've thought about it a bit more. Considering different ways `Link.FromActivity` should behave (use parent of new context), I want to remove this API at all. And call `Link.FromSpanContext` that does not have this ambiguity. If we'll ever need `FromActivity` we'll return it back in some form.

So here, would you mind explicitly calling `Link.FromSpanContext(SpanContext.Create(link.TraceId, link.ParentSpanId, link.ActivityTraceFlags, Tracestate.Empty))` ?